src/code/audio/code_801A5BD0.c

```diff
@@ -1,45 +1,780 @@
+        sSfxBankFreeListStart[bankId] = gSfxBanks[bankId][sSfxBankFreeListStart[bankId]].next;
+        gSfxBanks[bankId][sSfxBankFreeListStart[bankId]].prev = 0xFF;
+        entry->next = 0xFF;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_801A5BD0/func_801A794C.s")
+void Audio_RemoveSfxBankEntry(u8 bankId, u8 entryIndex) {
+    SfxBankEntry* entry = &gSfxBanks[bankId][entryIndex];
+    u8 i;
+
+    if (entry->sfxParams & 8) {
+        Audio_ClearFlagForBgmVolumeLow(entry->channelIdx);
+    }
+    if (entryIndex == sSfxBankListEnd[bankId]) {
+        sSfxBankListEnd[bankId] = entry->prev;
+    } else {
+        gSfxBanks[bankId][entry->next].prev = entry->prev;
+    }
+    gSfxBanks[bankId][entry->prev].next = entry->next;
+    entry->next = sSfxBankFreeListStart[bankId];
+    entry->prev = 0xFF;
+    gSfxBanks[bankId][sSfxBankFreeListStart[bankId]].prev = entryIndex;
+    sSfxBankFreeListStart[bankId] = entryIndex;
+    entry->state = SFX_STATE_EMPTY;
+
+    for (i = 0; i < gChannelsPerBank[gSfxChannelLayout][bankId]; i++) {
+        if (gActiveSfxs[bankId][i].entryIndex == entryIndex) {
+            gActiveSfxs[bankId][i].entryIndex = 0xFF;
+            i = gChannelsPerBank[gSfxChannelLayout][bankId];
+        }
+    }
+}
+
+void Audio_ChooseActiveSfxs(u8 bankId) {
+    u8 numChosenSfxs;
+    u8 numChannels;
+    u8 entryIndex;
+    u8 i;
+    u8 j;
+    u8 k;
+    u8 sfxImportance;
+    u8 needNewSfx;
+    u8 chosenEntryIndex;
+    u16 temp3;
+    s32 pad;
+    SfxBankEntry* entry;
+    ActiveSfx chosenSfxs[MAX_CHANNELS_PER_BANK];
+    ActiveSfx* activeSfx;
+    f32 tempf1;
+    f32 tempf2;
+
+    numChosenSfxs = 0;
+    for (i = 0; i < MAX_CHANNELS_PER_BANK; i++) {
+        chosenSfxs[i].priority = 0x7FFFFFFF;
+        chosenSfxs[i].entryIndex = 0xFF;
+    }
+    entryIndex = gSfxBanks[bankId][0].next;
+    k = 0;
+    while (entryIndex != 0xFF) {
+        if ((gSfxBanks[bankId][entryIndex].state == SFX_STATE_QUEUED) &&
+            (gSfxBanks[bankId][entryIndex].sfxId & 0xC00)) {
+            gSfxBanks[bankId][entryIndex].freshness--;
+        } else if (!(gSfxBanks[bankId][entryIndex].sfxId & 0xC00) &&
+                   (gSfxBanks[bankId][entryIndex].state == SFX_STATE_PLAYING_2)) {
+            Audio_QueueCmdS8((gSfxBanks[bankId][entryIndex].channelIdx << 8) | 0x6020000, 0);
+            Audio_RemoveSfxBankEntry(bankId, entryIndex);
+        }
+        if (gSfxBanks[bankId][entryIndex].freshness == 0) {
+            Audio_RemoveSfxBankEntry(bankId, entryIndex);
+        } else if (gSfxBanks[bankId][entryIndex].state != SFX_STATE_EMPTY) {
+            entry = &gSfxBanks[bankId][entryIndex];
+
+            if (&gSfxDefaultPosScreenCenter.x == entry[0].posX) {
+                entry->dist = 0.0f;
+            } else {
+                tempf1 = *entry->posY * 1;
+                tempf2 = *entry->posX * 0.5f;
+                entry->dist = (SQ(tempf2) + SQ(tempf1) + SQ(*entry->posZ)) / 10.0f;
```

Comment:
I'm not sure what the purpose of this is, but since this is the only place these are used, you can surely name them a bit more descriptively.